src/core/gosling-component.tsx

```diff
@@ -52,16 +52,16 @@ export const GoslingComponent = forwardRef<GoslingRef, GoslingCompProps>((props,
     const wrapperDivId = props.id ?? uuid.v4();
 
     // Gosling APIs
-    useEffect(() => {
-        if (!ref || !hgRef?.current) return;
-        const hgApi = hgRef.current;
-        const api = createApi(hgApi, viewConfig, trackInfos.current, theme);
-        if (typeof ref == 'function') {
-            ref({ api, hgApi });
-        } else {
-            ref.current = { api, hgApi };
-        }
-    }, [hgRef.current, viewConfig, theme]);
+    useImperativeHandle(
+        ref,
+        () => {
+            const hgApi = refAsReadonlyProxy(hgRef);
+            const infos = refAsReadonlyProxy(trackInfos);
+            const api = createApi(hgApi, viewConfig, infos, theme);
+            return { api, hgApi };
```

Comment:
I somewhat wonder if we should expose the low-level hgApi in Gosling.

Reply:
I think this can be useful when ones want to use API functions that Gosling does not yet support, but I generally agree that it's better not to expose higlass-specific features in Gosling. We can make this deprecated at some point.